src/ControlSystem/ControlErrors/Size/Update.hpp

```diff
@@ -0,0 +1,115 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <optional>
+#include <string>
+
+#include "ControlSystem/Averager.hpp"
+#include "ControlSystem/ControlErrors/Size.hpp"
+#include "ControlSystem/TimescaleTuner.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/Structure/ObjectLabel.hpp"
+#include "IO/Logging/Verbosity.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Printf.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace control_system::size {
+/*!
+ * \brief Updates the Averager with information from the \link
+ * control_system::ControlErrors::Size Size control error \endlink.
+ *
+ * \details After we calculate the control error, we check if a discontinuous
+ * change has occurred (the internal `control_system::size::State` changed). If
+ * no discontinuous change has occurred, then we do nothing. If one did occur,
+ * we get a history of control errors from the \link
+ * control_system::ControlErrors::Size Size control error \endlink and use that
+ * to repopulate the Averager history.
+ */
+template <size_t DerivOrder, ::domain::ObjectLabel Horizon,
+          typename Metavariables>
+void update_averager(
+    const gsl::not_null<Averager<DerivOrder - 1>*> averager,
+    const gsl::not_null<ControlErrors::Size<DerivOrder, Horizon>*>
+        control_error,
+    const Parallel::GlobalCache<Metavariables>& cache, const double time,
+    const DataVector& current_timescale,
+    const std::string& function_of_time_name, const int current_measurement) {
+  if (control_error->discontinuous_change_has_occurred()) {
+    control_error->reset();
+    averager->clear();
+
+    const std::deque<std::pair<double, double>> control_error_history =
+        control_error->control_error_history();
```

Comment:
Not really a change request, but here we need to be careful that the history obtained is the history of the new state, not the old state, if there has been a state change.

Reply:
The `control_error_history` function gets the history for the *current* state. The `udate_averager` function is called after we calculate the control error and potentially switch states so the control error should be in the *new* state. However, I'll add a note in `UpdateControlSystem` so that we are aware of this